tests/cbmc/proofs/s2n_array_init/s2n_array_init_harness.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_array.h"
+
+#include <assert.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/cbmc_utils.h>
+
+void s2n_array_init_harness()
+{
+    /* Non-deterministic inputs. */
+    uint32_t element_size;
+    struct s2n_array *array = can_fail_malloc(sizeof(struct s2n_array));
```

Comment:
@ttjsu-aws thanks for adding this proof. You don't need to use `can_fail_malloc` anymore. CBMC standard version of malloc will non-deterministically fail for all proofs on s2n. The following is perfectly fine,
```c
struct s2n_array *array = malloc(sizeof(*array));
```
The errors that you're seeing might be because you also need to allocate memory for the underlining pointer in `array->mem->data`. So I'd add the following,
```c
struct s2n_array *array = malloc(sizeof(*array));
if (array != NULL) array->mem->data = malloc(sizeof(*(array->mem->data))); 
```
At this point, `data` could either be NULL or properly allocated. So I'd also add another `ENSURE_REF` on `s2n_array_init` as 
```c
S2N_RESULT s2n_array_init(struct s2n_array *array, uint32_t element_size)
{
    ENSURE_REF(array);
    ENSURE_REF(array->mem->data);

    CHECKED_MEMSET(array, 0, sizeof(struct s2n_array));
    array->element_size = element_size;

   GUARD_RESULT(s2n_array_validate(array));
    return S2N_RESULT_OK;
}
```

Note that, at the end, I also added the validity check in the function, so we can make sure at runtime that the initialized array is a valid structure. I hope that helps.